src/common/components/repository-input/index.js

```diff
@@ -0,0 +1,70 @@
+import 'isomorphic-fetch';
+
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import {
+  changeRepositoryInput,
+  verifyGitHubRepository
+} from '../../actions/repository-input';
+
+export class RepositoryInput extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  getStatusMessage() {
+    const input = this.props.repositoryInput;
+    let message;
+
+    if (!input.repository) {
+      message = 'Invalid repository URL.';
+    } else if (input.repository && input.isFetching) {
+      message = `Verifying ${input.repository} on GitHub...`;
+    } else if (input.success && input.repositoryUrl) {
+      message = `Repository ${input.repository} is valid.`;
+    } else if (input.errors) {
+      message = `Repository ${input.repository} is invalid.`;
```

Comment:
Yes, messages are not very helpful ATM. Moving sync parsing/validation to input update function separately from calling GH API should help with that.